File: src/main/java/abstract_factory/product/part/OffloadVehiclePartFactory.java
package abstract_factory.product.part;

import java.util.List;

public class OffloadVehiclePartFactory implements VehiclePartFactory {
    @Override
    public Engine createEngine() {
        return Engine.GASOLINE;
    }

    @Override
    public Tire createTire() {
        return Tire.OFFLOAD;
    }

    @Override
    public List<Option> createOptions() {
        return List.of(Option.NAVIGATION, Option.AUDIO_SYSTEM);
    }
}

File: kotlin/src/main/kotlin/abstract_factory/product/part/OffloadVehiclePartFactory.kt
package abstract_factory.product.part

class OffloadVehiclePartFactory : VehiclePartFactory {
    override fun createEngine(): Engine {
        return Engine.GASOLINE
    }

    override fun createTire(): Tire {
        return Tire.OFFLOAD
    }

    override fun createOptions(): List<Option> {
        return listOf(Option.NAVIGATION, Option.AUDIO_SYSTEM)
    }
}